serde/src/de/mod.rs

```diff
@@ -568,6 +568,11 @@ pub trait VariantVisitor {
         Err(Error::syntax_error())
     }
 
+    /// `visit_simple` is called when deserializing a variant with a single value.
+    fn visit_simple<T: Deserialize>(&mut self) -> Result<T, Self::Error> {
+        Err(Error::syntax_error())
```

Comment:
instead of erroring this could try to do a `visit_seq` by default to get back the original behavior for downstream `Deserializer` implementors